vm/src/compiler.rs

```diff
@@ -312,14 +322,16 @@ impl<'a, T: CompilerEnv> CompilerEnv for &'a T {
 impl<'a> Compiler<'a> {
     pub fn new(globals: &'a (CompilerEnv + 'a),
                vm: &'a GlobalVmState,
-               symbols: SymbolModule<'a>)
+               symbols: SymbolModule<'a>,
+               source_map: &'a Source<'a>)
```

Comment:
Yeah... it's interesting that they are similar in function. Let's see how it pans out - maybe in the future we can find some common idea in `Source` and `SourceMap` to make this stuff simpler.